src/main/java/Duke.java

```diff
@@ -1,10 +1,167 @@
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.stage.Stage;
+import java.io.FileNotFoundException;
+import java.time.format.DateTimeParseException;
+
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+    // Empty constructor required for Launcher
+    public Duke() {
+    }
+
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTasksFromSaveFile());
+
+            // Since we can successfully load it, we save all in TaskList
+            storage.writeTasks(tasks);
+        } catch (FileNotFoundException e) {
+            // Load an empty one
+            tasks = new TaskList();
+            Ui.printLine();
+            Ui.indent("It seems that there is no save file in: " + filePath);
+            Ui.printLine();
+        }
+    }
+
+    public void run() {
+        Ui.greet();
+        while (Parser.stillHasInputs()) {
+            try {
+                String input = Parser.readLine();
+                if (Parser.commandEquals("bye", input)) {
+                    Ui.sayGoodbye();
+                    break;
+                } else if (Parser.commandEquals("list", input)) {
+                    Ui.displayList(tasks);
+                } else if (Parser.commandEquals("done", input)) {
+                    int taskNo = Parser.getTaskNo(input);
+                    tasks.markTaskAsDone(taskNo);
+                } else if (Parser.commandEquals("delete", input)) {
+                    int taskNo = Parser.getTaskNo(input);
+                    tasks.deleteTask(taskNo);
+                } else if (Parser.commandEquals("deadline", input)) {
+                    if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
```

Comment:
minor issue. do note that if-else blocks have to be covered in curly braces, according to this mod's coding standard